boards/sparkfun/micromod/doc/index.rst

```diff
@@ -0,0 +1,212 @@
+| USB       | on-chip    | usb                                 |
++-----------+------------+-------------------------------------+
+| QSPI      | on-chip    | nordic,nrf-qspi                     |
++-----------+------------+-------------------------------------+
+
+
+Connections and IOs
+===================
+
+LED
+---
+
+* Led0 (blue) = P0.13
+
+Micromod header
+---------------
+
+* micromod_1_uart = uart0
+* micromod_2_uart = uart1
+* micromod_0_i2c = i2c0
+* micromod_1_i2c = i2c1
+* micromod_0_spi = spi2
+* micromod_header compatible with ``sparkfun,micromod-gpio``
+
+Programming and Debugging
+*************************
+
+Applications for the ``micromod/nrf52840`` board target can be
+built, flashed, and debugged in the usual way. See
+:ref:`build_an_application` and :ref:`application_run` for more details on
+building and running.
+
+Flashing
+========
+
+Follow the instructions in the :ref:`nordic_segger` page to install
+and configure all the necessary software. Further information can be
+found in :ref:`nordic_segger_flashing`. Then build and flash
+applications as usual (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+The flashing tool will depend on the carrier used along with the board.
+In the case of `Sparkfun asset tracking carrier`, it is possible to use
+the SWD interface along with a J-Link.
+
+Here is an example for the :ref:`hello_world` application.
+
+First, run your favorite terminal program to listen for output.
+
+.. code-block:: console
+
+   $ minicom -D <tty_device> -b 115200
+
+Replace :code:`<tty_device>` with the port where the board nRF52840 DK
+can be found. For example, under Linux, :code:`/dev/ttyACM0`.
+
+Then build and flash the application in the usual way.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: micromod/nrf52840
+   :goals: build flash
+
+Debugging
+=========
+
+Refer to the :ref:`nordic_segger` page to learn about debugging Nordic boards with a
+Segger IC.
+
+In case of using a Nordic Segger Jlink, it is possible to configure the app to use Segger Real
+Time Transfer (RTT) for debugging. In this case, build your app with the
+:ref:`RTT snippet <snippet-rtt-console>`.
+
+Testing the on-board LED
+************************
+
+There is a sample that allow you to test the LED on the board is working properly
+with Zephyr:
+
+* :zephyr:code-sample:`blinky`
+
+You can build and flash the example to make sure Zephyr is running correctly on
+your board. The button and LED definitions can be found in
+:zephyr_file:`boards/sparkfun/micromod/micromod_nrf52840.dts`.
+
+Testing the 128Mb qspi memory flash
+***********************************
+
+The ``micromod/nrf52840`` board target has a built-in NOR flash memory connected
+to the qspi interface. It can be tested with the samples/drivers/spi_flash app.
+
+Add the config overlay file to the boards directory of the sample
+
+   .. code-block:: kconfig
```

Comment:
"kconfig" is for actual kconfig file, not ".config" files

```suggestion
   .. code-block:: cfg
```

Reply:
I have modified this section a bit, because the best sample is the `/samples/drivers/jesd216` app instead of the `spi_flash` sample since it is a QSPI flash. I got the samples mixed. I checked it out and works with the jesd216 sample smoothly :D